include/pybind11/pytypes.h

```diff
@@ -380,20 +383,49 @@ class PYBIND11_EXPORT_EXCEPTION error_already_set : public std::runtime_error {
 public:
     /// Constructs a new exception from the current Python error indicator, if any.  The current
     /// Python error indicator will be cleared.
-    error_already_set() : std::runtime_error(detail::error_string()) {
+    error_already_set() : std::runtime_error("") {
         PyErr_Fetch(&m_type.ptr(), &m_value.ptr(), &m_trace.ptr());
+        if (m_type) {
+            PyErr_NormalizeException(&m_type.ptr(), &m_value.ptr(), &m_trace.ptr());
+            /*if (m_trace) {
```

Comment:
Causes errors on PyPy interestingly enough.

Reply:
#ifdef?

Reply:
It doesn't match the old behavior we had though; I don't want to break anything. And it's additional overhead that we aren't using.